numpy/core/tests/test_ufunc.py

```diff
@@ -287,18 +287,85 @@ def test_all_ufunc(self):
         pass
 
     def test_signature(self):
+        # from include/numpy/ufuncobject.h
```

Comment:
Maybe just a question of style, but I'd probably favor splitting the 78 line unit test function here into separately named test functions such that when they fail there's a more meaningful connection between the failure and the name of the test.

Parametrizing the tests from there may also further assist with meaningful test error tracebacks, but arguably this is just bonus stuff I suppose & these big "unit tests" are fairly common from what I've seen so far in NumPy source.

Reply:
Split them into separate tests